File: WMS.EntityCore/Cliente/clsBeCliente_tiempos.cs
using WMS.EntityCore.Producto;

namespace WMS.EntityCore.Cliente
{
    public class clsBeCliente_tiempos : ICloneable, IDisposable
    {
        public int IdTiempoCliente { get; set; } = 0;
        public int IdCliente { get; set; } = 0;
        public int IdFamilia { get; set; } = 0;
        public int IdClasificacion { get; set; } = 0;
        public clsBeProducto_familia Familia { get; set; }= new clsBeProducto_familia();
        public clsBeProducto_clasificacion Clasificacion { get; set; } = new clsBeProducto_clasificacion();
        public int Dias_Local { get; set; } = 0;
        public int Dias_Exterior { get; set; } = 0;
        public string User_agr { get; set; } = "";
        public DateTime Fec_agr { get; set; } = DateTime.Now;
        public string User_mod { get; set; } = "";
        public DateTime Fec_mod { get; set; } = DateTime.Now;
        public bool Activo { get; set; } = false;

        public clsBeCliente_tiempos()
        {
        }

        public clsBeCliente_tiempos(int IdTiempoCliente, int IdCliente, int IdFamilia, int IdClasificacion,
                                   int Dias_Local, int Dias_Exterior, string user_agr, DateTime fec_agr,
                                   string user_mod, DateTime fec_mod, bool activo)
        {
            this.IdTiempoCliente = IdTiempoCliente;
            this.IdCliente = IdCliente;
            this.IdFamilia = IdFamilia;
            this.IdClasificacion = IdClasificacion;
            this.Dias_Local = Dias_Local;
            this.Dias_Exterior = Dias_Exterior;
            this.User_agr = user_agr;
            this.Fec_agr = fec_agr;
            this.User_mod = user_mod;
            this.Fec_mod = fec_mod;
            this.Activo = activo;
        }

        public object Clone()
        {
            return this.MemberwiseClone();
        }

        #region IDisposable Support
        private bool disposedValue = false;

        protected virtual void Dispose(bool disposing)
        {
            if (!disposedValue)
            {
                if (disposing)
                {
                    // TODO: dispose managed state (managed objects).
                }

                // TODO: free unmanaged resources (unmanaged objects) and override finalizer
                // TODO: set large fields to null
                disposedValue = true;
            }
        }

        // // TODO: override finalizer only if 'Dispose(bool disposing)' has code to free unmanaged resources
        // ~clsBeCliente_tiempos()
        // {
        //     // Do not change this code. Put cleanup code in 'Dispose(bool disposing)' method
        //     Dispose(disposing: false);
        // }

        public void Dispose()
        {
            Dispose(true);
            GC.SuppressFinalize(this);           
        }
        #endregion
    }
}

File: TOMIMSV4/Entity/Mantenimientos/Cliente/clsBeCliente_tiempos.vb
Public Class clsBeCliente_tiempos
    Implements ICloneable
    Implements IDisposable

    Public Property IdTiempoCliente() As Integer = 0
    Public Property IdCliente() As Integer = 0
    Public Property IdFamilia() As Integer = 0
    Public Property IdClasificacion() As Integer = 0
    Public Property Familia As clsBeProducto_familia
    Public Property Clasificacion As clsBeProducto_clasificacion
    Public Property Dias_Local() As Integer = 0
    Public Property Dias_Exterior() As Integer = 0
    Public Property User_agr() As String = ""
    Public Property Fec_agr() As Date = Date.Now
    Public Property User_mod() As String = ""
    Public Property Fec_mod() As Date = Date.Now
    Public Property Activo() As Boolean = False

    Public Property Es_Manufactura As Boolean = False

    Sub New()
    End Sub

    Sub New(ByRef IdTiempoCliente As Integer, ByVal IdCliente As Integer, ByVal IdFamilia As Integer, ByVal IdClasificacion As Integer, ByVal Dias_Local As Integer, ByVal Dias_Exterior As Integer, ByVal user_agr As String, ByVal fec_agr As Date, ByVal user_mod As String, ByVal fec_mod As Date, ByVal activo As Boolean)
        Me.IdTiempoCliente = IdTiempoCliente
        Me.IdCliente = IdCliente
        Me.IdFamilia = IdFamilia
        Me.IdClasificacion = IdClasificacion
        Me.Dias_Local = Dias_Local
        Me.Dias_Exterior = Dias_Exterior
        Me.User_agr = user_agr
        Me.Fec_agr = fec_agr
        Me.User_mod = user_mod
        Me.Fec_mod = fec_mod
        Me.Activo = activo
    End Sub
    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function

#Region "IDisposable Support"
    ''' <summary>
    ''' The disposed value
    ''' </summary>
    Private disposedValue As Boolean ' To detect redundant calls

    ' IDisposable
    ''' <summary>
    ''' Releases unmanaged and - optionally - managed resources.
    ''' </summary>
    ''' <param name="disposing"><c>true</c> to release both managed and unmanaged resources; <c>false</c> to release only unmanaged resources.</param>
    Protected Overridable Sub Dispose(disposing As Boolean)
        If Not disposedValue Then
            If disposing Then
                ' TODO: dispose managed state (managed objects).
            End If

            ' TODO: free unmanaged resources (unmanaged objects) and override Finalize() below.
            ' TODO: set large fields to null.
        End If
        disposedValue = True
    End Sub

    ' TODO: override Finalize() only if Dispose(disposing As Boolean) above has code to free unmanaged resources.
    'Protected Overrides Sub Finalize()
    '    ' Do not change this code.  Put cleanup code in Dispose(disposing As Boolean) above.
    '    Dispose(False)
    '    MyBase.Finalize()
    'End Sub

    ' This code added by Visual Basic to correctly implement the disposable pattern.
    ''' <summary>
    ''' Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
    ''' </summary>
    Public Sub Dispose() Implements IDisposable.Dispose
        ' Do not change this code.  Put cleanup code in Dispose(disposing As Boolean) above.
        Dispose(True)
        ' TODO: uncomment the following line if Finalize() is overridden above.
        ' GC.SuppressFinalize(Me)
        If Familia IsNot Nothing Then
            Familia.Dispose()
            Familia = Nothing
        End If
    End Sub
#End Region

End Class
